### PR TITLE
CCS WLH variant and driver question

### PR DESCRIPTION
**Pull request by petechd**
### What is the context of this PR?

This adds new variants to _anyone-else-list-collector_ blocks based on address and household list rules. There is new driving question variant and _add_block_ variants. _Add_answer_ block has to accept new values so local variable for different answer label was used.

### How to review

Build schemas and check the journey or use links below.

#### England

- [CCS](https://test-launcher.gcp.dev.eq.ons.digital/quick-launch?language_code=en&survey=CENSUS&form_type=I&region_code=GB-ENG&url=https://storage.googleapis.com/eq-questionnaire-schemas-artifacts/wlh-variant-and-driver-question/schemas/en/ccs_household_gb_eng.json)

#### Wales

- [CCS](https://test-launcher.gcp.dev.eq.ons.digital/quick-launch?language_code=en&survey=CENSUS&form_type=I&region_code=GB-ENG&url=https://storage.googleapis.com/eq-questionnaire-schemas-artifacts/wlh-variant-and-driver-question/schemas/en/ccs_household_gb_wls.json)

### CODE REVIEW

**Comment by petechd** [comment]
[(I) CCS- Create WLH variant and add driver question (S)](https://trello.com/c/X9QHEPtq/3958-i-ccs-create-wlh-variant-and-add-driver-question-s)

File: source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_driver.jsonnet [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      ],
+    },
+    type: 'General',
+    answers: [
+      {
+        id: 'anyone-else-driver-answer',
+        mandatory: false,
+        options: [
+          {
+            label: 'Ok, understood',
```
Comment (MebinAbraham) [inline source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_driver.jsonnet:22]: On overflow it's `OK`.

File: source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet [hunk 2]
```diff
@@ -182,37 +181,98 @@ local editQuestion(questionTitle) = {
...
+        question: {
+          id: 'add-question',
+          type: 'General',
+          title: {
+            text: 'Who do you need to add?',
+            placeholders: [
+              placeholders.address,
+            ],
+          },
+          instruction: 'Enter a full stop (.) if the respondent does not know a person’s “First name” or “Last name”',
```
Comment (MebinAbraham) [inline source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet:251]: On overflow, this is also shown for the first list collector. Can you verify what should be the correct behaviour?

Comment (petechd) [reply source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet:251]: I followed the card:
"New driver question to route to a **new variant add answer screen** on anyone-else-list collector (New variant- 'What is your full name?'). **New variant should not have orange instruction panel**.

Variant when list empty- 'What is your full name?'
Variant when list not empty- 'Who do you need to add?'"

Did you check [overflow](https://overflow.io/s/FQ1RSI?node=3bf45946) attached to this card?

File: source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_driver.jsonnet (hunk 1 above)
Comment (petechd) [reply source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_driver.jsonnet:22]: It's 'Ok, understood' on[ overflow](https://overflow.io/s/FQ1RSI?node=3bf45946) attached to this card.

File: source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet (hunk 2 above)
Comment (MebinAbraham) [reply source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet:251]: The new variant shouldn't have it and that is correct but for the variant `Who do you need to add?` it should be displayed according to Overflow. On the quick launch, for the question `Who do you need to add?` the panel isn't displayed.

Comment (petechd) [reply source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet:251]: Ok, I thought it is a new variant with two options, one of them is "when list not empty- 'Who do you need to add?'"?? I'll check with Donna.

File: source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_driver.jsonnet (hunk 1 above)
Comment (MebinAbraham) [reply source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_driver.jsonnet:22]: Sorry, I should have been more explicit, it's `OK, understood` on overflow.
image

File: source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet (hunk 2 above)
Comment (petechd) [reply source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet:247]: Done, thanks!👍

File: source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_driver.jsonnet (hunk 1 above)
Comment (petechd) [reply source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_driver.jsonnet]: Sorted, thanks!👍

File: source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_driver.jsonnet [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      text: 'I will now collect the names of everyone who lived at that address on Sunday {census_date}',
+      placeholders: [
+        placeholders.censusDate,
+      ],
+    },
+    type: 'General',
+    answers: [
+      {
+        id: 'anyone-else-driver-answer',
+        mandatory: false,
```
Comment (MebinAbraham) [inline source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_driver.jsonnet:19]: This should be made mandatory, as we cannot route to the add screen when they haven't answered. Already checked with Donna.

Comment (petechd) [reply source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_driver.jsonnet]: Done, thanks!👍

File: source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet [hunk 4]
```diff
@@ -182,37 +181,94 @@ local editQuestion(questionTitle) = {
...
+              id: 'last-name',
+              label: 'Last name',
+              mandatory: true,
+              type: 'TextField',
+            },
+          ],
         },
-      ],
-    },
+        when: [rules.livesAtHouseholdAddress],
```
Comment (MebinAbraham) [inline source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet:269]: Not all 3 are needed?. According to overflow, it should just be a variant when list is empty and when list is not empty. The variant for `[rules.listIsNotEmpty('household'), rules.doesntLiveAtHouseholdAddress]` and `[rules.livesAtHouseholdAddress]` are identical.

Comment (MebinAbraham) [reply source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet:269]: Actually, it looks like a consequence of us not allowing "or" in variants. We should split the variant into a partial in jsonnet to reduce the duplication.

Comment (petechd) [reply source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet]: Ok, I removed extra variant, thanks!👍

File: source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet [hunk 5]
```diff
@@ -182,37 +181,65 @@ local editQuestion(questionTitle) = {
...
+              type: 'TextField',
+            },
+            {
+              id: 'last-name',
+              label: 'Last name',
+              mandatory: true,
+              type: 'TextField',
+            },
+          ],
         },
```
Comment (MebinAbraham) [inline source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet:210]: We should make this a partial and pass in the title, the rest is identical. Same way the edit block is done.

https://github.com/ONSdigital/eq-questionnaire-schemas/blob/wlh-variant-and-driver-question/source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet#L244-L257

Comment (petechd) [reply source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet:210]: The only difference is lack of `instruction` block in one of the variants.

Comment (MebinAbraham) [reply source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet:210]: Ya, so we should be able to deduplicate the rest and add an instruction key to the single variant.

**Pull request merged**